tests/unit_tests/test_tools.py

```diff
@@ -136,3 +139,38 @@ def test_select_rec_for_thumbnail():
     assert (
         select_rec_for_thumbnail(rec_names, thumbnail_rec="sAHP_20.soma.v") == "IDrest_130.soma.v"
     )
+
+
+def test_get_protocol_name():
```

Comment:
In the tests could you also try with strings from response keys (i.e. without the feature at the end, e.g. 'IV_40.0.soma.v')